qemu/tests/ksm_ksmtuned.py

```diff
@@ -83,23 +82,24 @@ def check_qemu_used_mem(qemu_pid, mem):
                                    stress_timeout, 10, 10):
             test.error("QEMU used memory doesn't reach %s of guest mem %sM in "
                        "%ss" % (mem_thres, mem // 1024, stress_timeout))
-        time.sleep(30)
+        cmd = params['cmd_check_ksm_status']
         free_mem_host = utils_memory.freememtotal()
-        ksm_status = process.getoutput(params['cmd_check_ksm_status'])
+        ksm_status = utils_misc.wait_for(lambda: '1' == process.getoutput(cmd),
+                                         60)
```

Comment:
Base on your test result,  30s is not enough sometimes but can work most of the time, suggest add `first` param which means `Time to sleep before first attempt`, for example `first = 20`

Reply:
@yanan-fu IIUC I need to add a sleep with the first value and reduce the wait_for till 30s again?

Reply:
@mcasquer  I mean the `first` param for `wait_for` function

Reply:
refer:https://github.com/avocado-framework/avocado-vt/blob/2ddf56c099e314351d624bc33d6a7289f33a9804/virttest/utils_misc.py#L543

Reply:
@yanan-fu when replacing the function call to:
`utils_misc.wait_for(lambda: '1' == process.getoutput(cmd), 40, first=20.0)`
The test case fails complaining about KSM should be running
```
 (1/1) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned.q35: STARTED
 (1/1) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned.q35: FAIL: KSM should be running (258.96 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```
Any idea what's the difference that make the same test pass with:
`utils_misc.wait_for(lambda: '1' == process.getoutput(cmd), 60)`

Reply:
After trying on a different host, timeout=40s and first=20.0 works fine

Reply:
```
 (1/5) repeat1.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned.q35: STARTED
 (1/5) repeat1.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned.q35: PASS (290.78 s)
 (2/5) repeat2.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned.q35: STARTED
 (2/5) repeat2.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned.q35: PASS (286.98 s)
 (3/5) repeat3.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned.q35: STARTED
 (3/5) repeat3.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned.q35: PASS (298.66 s)
 (4/5) repeat4.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned.q35: STARTED
 (4/5) repeat4.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned.q35: PASS (296.60 s)
 (5/5) repeat5.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned.q35: STARTED
 (5/5) repeat5.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned.q35: PASS (296.76 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```

Reply:
@yanan-fu do you agree with these values?

Reply:
Ok for me, thanks!